### ISSUE TITLE
Julia setup failure and (also) setup succcess !!

### ISSUE THREAD

**Issue by elemeur** [Non-Contradiction/JuliaCall#119]
Hello, I had the same problem as issue #112.[failure by setup,same message and also some brutal hanging]. But I also had success setting up JuliaCall on another install of R on the same machine...I can post the logs if needed.
All in all, I have 3.5.1, 3.5.2, 3.5.3 and 3.6.1. Only 3.5.2 works.
That installation is special : it is referenced in my (windows) path variable and is also the target for such instructions as R CMD install. 
Does that help?

**Comment by Non-Contradiction** [comment]
Thanks for the feedback.
The problem of using JuliaCall with different versions of R is a little complicated. It really depends on how you install R, JuliaCall, and JuliaCall dependencies... 
First, there is a `rebuild` argument for `julia_setup`, and it needs to be set to `TRUE` for the first time of using JuliaCall with a new R installation. Hopefully, this can solve some of the problems with different installation of R, but there are situations it doesn't work...
If things do not work, then there could be other solutions. For R installation not in PATH, you need to find `R.home()` from the R you want to use, and set `ENV["R_HOME"]="the path you just get from R.home()"` in Julia, and run `using Pkg; Pkg.build("RCall")`.

**Comment by BabaYara** [comment]
Hi Non-Contradiction, 

I think you have to include your last post on this thread as part of the installation. 

I have been having a hard time getting julia_setup() to run all week. (julia_setup() was always hanging after  "Loading setup script for JuliaCall..."
I only managed to get it to run after implementing your suggestion with the R.home() and Julia environment setup and rebuild of RCall.. 

It will definitely save people a lot of headaches. 
I am using Rstudio on windows 10 with R 4.03

**Comment by FridljDa** [comment]
Hi BabaYara,

Thank you for giving the comment! Could you perhaps recapitulate step by step what you did, to solve this issue?

**Comment by BabaYara** [comment]
Open your new Julia repl.
1. Run: ENV["R_HOME"]="the path you just get from R.home()"
  You can find your R home by running R.home() inside R
  Once you are done with step 1. 
2. Run: using Pkg; Pkg.build("RCall")  
in repl. 
Once Julia is done building and compiling the new RCall. 

You should be able to rebuild inside R.

**Comment by Faraz564** [comment]
AOA Baba Yara
I have open a new issue $201 with subject to juliacall. I also tried your method but could not sussessful to resolve this issue. 
Actually I have problem running Juliacall syntax in console while julia can traced in terminal.
Plz see it below in picture and suggest me a solution.
![JuliaCall](https://github.com/Non-Contradiction/JuliaCall/assets/116310732/32637ab2-e66d-4559-92aa-aa9385ca44a6)
